### PR TITLE
Pass watchedNodes to Watcher/WatcherAdapter

### PR DESCRIPTION
**Pull request by oligriffiths**
The intention of this PR is to make the `watchedNodes` an argument to the constructor of the watcher, as they're a required dependency and watcher can't work without them.
Additionally, NOT reaching into the builder to get them will help with the ember CLI integration of the watcher.

### CODE REVIEW

**Comment by thoov** [comment]
@oligriffiths Should this be apart of the 3.0 release or should it come after?

**Comment by oligriffiths** [comment]
@thoov I’d prefer if it were but I’m not going to be able to work much more on it until Monday

**Comment by lifeart** [comment]
Just wondering, if it's a have sence - use current git revision for trigger rebuild if it changes (for upstream changes) or use requests from IDE like VSCode language server to get exact changed files (without watching overhead)

**Comment by oligriffiths** [comment]
@lifesrt the fit revision doesn’t know about changed files, only committed ones. Not sure how this would help?

**Comment by thoov** [comment]
@oligriffiths This looks good. Just to confirm that this is so we dont have to do this? (https://github.com/ember-cli/ember-cli/pull/8614/files#diff-9ff56008b40607a31bc04df7f41906cbR45)

File: test/watcher_adapter_test.js [hunk 1]
```diff
@@ -88,11 +113,45 @@ describe('WatcherAdapter', function() {
...
+      expect(() => new WatcherAdapter([new TransformNodeWrapper()])).to.throw(
+        Error,
+        `[NodeWrapper:undefined undefined at undefined] is not a SourceNode`
+      );
+    });
+
+    it('throws if you try to watch a non-watchable node', function() {
+      expect(() => new WatcherAdapter([unwatchedNodeBasic])).to.throw(
+        Error,
+        `/Users/goli/Projects/broccoli/test/fixtures/basic it not watched`
```
Comment (stefanpenner) [inline test/watcher_adapter_test.js]: hard coded local path

Comment (oligriffiths) [reply test/watcher_adapter_test.js]: lol, oh yeah

**Comment by oligriffiths** [comment]
@thoov correct. Also, the things to watch (nodes) are a direct dependency of this class. To me it makes more sense for them to be a mandatory argument rather than reaching in to the builder to get them

File: test/watcher_test.js [hunk 2]
```diff
@@ -199,7 +202,7 @@ describe('Watcher', function() {
 
     it('does nothing if already quitting', function() {
       const adapterQuit = sinon.spy(adapter, 'quit');
-      const watcher = new Watcher(builder, { watcherAdapter: adapter });
+      const watcher = new Watcher(builder, [watchedNodeBasic], { watcherAdapter: adapter });
       watcher._quittingPromise = true;
```
Comment (stefanpenner) [inline test/watcher_test.js]: using private API in tests is an anti pattern, let's try and use public API's

Comment (oligriffiths) [reply test/watcher_test.js]: Sure thing

**Comment by stefanpenner** [comment]
@lifeart I think asking git, for what is different, isn't a bad idea. We should explore that sometime.

**Comment by oligriffiths** [comment]
@rwjblue valid points, will fix

**Pull request merged**